File: lib-requirement-code/src/main/java/ir/adnan/lib_requirement_code/main/LibraryApplication.java
package ir.adnan.lib_requirement_code.main;

/**
 * Created by Adnan on 9/18/2017.
 */
import android.app.Application;
import android.content.Context;

import com.crashlytics.android.Crashlytics;

import io.fabric.sdk.android.Fabric;

public class LibraryApplication extends Application {

    private static Context THIS_APPLICATION = null;

    @Override
    public void onCreate() {
        super.onCreate();
        THIS_APPLICATION = this;
    }

    public static Context  getContext () {
        return THIS_APPLICATION;
    }

    protected void launchFabricCrashlytics (Context context , boolean debuggable) {

        final Fabric fabric = new Fabric.Builder(context)
                .kits(new Crashlytics())
                .debuggable(debuggable)
                .build();
        Fabric.with(fabric);
    }

}

File: lib-requirement-code/src/main/java/ir/adnan/lib_requirement_code/main/LibraryApplication.kt
package ir.adnan.lib_requirement_code.main

import android.app.Application
import android.content.Context

import com.crashlytics.android.Crashlytics

import io.fabric.sdk.android.Fabric


/**
 * Created by Adnan on 7/2/2017.
 */

open class LibraryApplication : Application() {

    companion object {

        var context: Context? = null
            @JvmStatic get() = this.context
    }

    override fun onCreate() {
        super.onCreate()
        context = this

//        launchDaggerRetrofit()
    }

    protected fun launchFabricCrashlytics(context: Context, debuggable: Boolean) {

        val fabric = Fabric.Builder(context)
                .kits(Crashlytics())
                .debuggable(debuggable)
                .build()
        Fabric.with(fabric)
    }
}
